src/core/include/nodes.h

```diff
@@ -32,6 +32,7 @@ namespace evoplex {
 class Nodes : private std::unordered_map<int, Node>
 {
     friend class AbstractGraph;
+    friend class GraphView;
```

Comment:
The `GraphView` class belongs to the `EvoplexGUI` project., while the `Nodes` class belongs to the `EvoplexCore` project. So, adding it here is a bad practice as the core should be kept completely untied to the gui stuff.

I see that you probably did it to access the `insert` function, right?

You may be wondering why the `Nodes` class doesn't support insertion.
ANS: because `Nodes` is part of the API exposed to the plugins (models and graphs), and while Nodes is just a container, if we make the `insert` function available, modellers might think that this is the right way to add a node to the graph. However, only the graph should be able to add a node to itself...

So, in the case of the `m_selectedNodes` member, it should be defined as a simple `std::vector<Node>`

Reply:
Ah I see how this could be a problem. I replaced this with a simple ``std::vector`` so it should be ok now.